### DISCUSSION TITLE
Call for users

### DISCUSSION THREAD

**Discussion by tomasherman** [datadog4s/datadog4s#509] [General]
Hello, 

I'd like to ask for the users for datadog4s(dd4s) to share some information about how they use it. It would help me to figure out how to best move forward with this library. I'd love the know the following:

- what cats-effect version of dd4s do you use (ce2 or ce3?)
- do you use http4s module? Are there other modules you would like to see?
- are you happy with the library? Is there something that would make your life easier?
- optionally: if you are using dd4s in a company, what company do you use it for?

Thanks in advance to everyone who will contribute any info :)

**Comment by josevicenteayala** [comment]
@tomasherman Hello Tomas, currently I'm planning to use dd4s with http4s to report APM (Application Performance Monitoring) metrics into DataDog in a Retail company, but I'm looking forward if you have more documentation/examples to understand how to use dd4s specifically the http4s module. I've been checking the [userguide](https://avast.github.io/datadog4s/userguide.html) link but I'm not fully understand how to report the APM metrics. 

Thanks for your help.

**Reply by tomasherman** [reply]
Hi, thanks for the interest/question - unfortunately dd4s doesnt provide much (any) support for ddog tracing. 

IIRC, datadog supports opentracing API, so theoretically you can find a http4s opentracing implementation (it seems there are some around based on quick google serach) and then configure the opentracing to use datadog agent for reporting. We had some success with that historically. Unfortunately the project where we used it has been closed so i don't have a lot of details on top of mind.

Good luck
